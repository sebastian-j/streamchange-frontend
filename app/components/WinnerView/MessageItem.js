import React from 'react';
import PropTypes from 'prop-types';

function MessageItem(props) {
  const dt = new Date(props.date);
  const convertedDate = `${dt.getHours()}:${
    dt.getMinutes() < 10 ? '0' : ''
  }${dt.getMinutes()}:${dt.getSeconds() < 10 ? '0' : ''}${dt.getSeconds()}`;
  return (
    <li className="winner-message">
      <span>{convertedDate}</span>
      <span>{props.text}</span>
    </li>
  );
}

MessageItem.propTypes = {
  date: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default MessageItem;
