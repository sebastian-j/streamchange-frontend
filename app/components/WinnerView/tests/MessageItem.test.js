import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import MessageItem from '../MessageItem';

const renderer = new ShallowRenderer();

describe('<MessageItem />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<MessageItem date="2019-12-24T07:27:56.27" text="text" />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
