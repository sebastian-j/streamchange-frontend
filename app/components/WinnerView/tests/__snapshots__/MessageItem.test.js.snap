// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageItem /> should render and match the snapshot 1`] = `
<ForwardRef>
  <ForwardRef>
    7:27:56
  </ForwardRef>
  <ForwardRef>
    text
  </ForwardRef>
</ForwardRef>
`;
